streaming/java/streaming-state/src/main/java/org/ray/streaming/state/backend/TransactionKeyStateBackend.java

```diff
@@ -0,0 +1,193 @@
+  protected Map<String, MapStateFacade> mapStateMngMap = new HashMap<>();
+  protected Set<String> descNamespace;
+
+  /**
+   * tablename, IKVStore key, checkpointId, content
+   */
+  protected Map<String, IKVStore<String, Map<Long, byte[]>>> backStorageMap;
+  private AbstractStateBackend backend;
+
+  public TransactionKeyStateBackend(AbstractStateBackend backend) {
+    this.backStorageMap = new HashMap<>();
+    this.backend = backend;
+    this.descNamespace = new HashSet<>();
+  }
+
+  public <K, T> void put(AbstractStateDescriptor descriptor, K key, T value) {
+    String desc = descriptor.getIdentify();
+    if (descriptor.getDescType() == DescType.value) {
+      if (this.valueStateMngMap.containsKey(desc)) {
+        valueStateMngMap.get(desc).put((String) key, value);
+      }
+    } else if (descriptor.getDescType() == DescType.list) {
+      if (this.listStateMngMap.containsKey(desc)) {
+        listStateMngMap.get(desc).put((String) key, (List) value);
+      }
+    } else if (descriptor.getDescType() == DescType.map) {
+      if (this.mapStateMngMap.containsKey(desc)) {
+        mapStateMngMap.get(desc).put((String) key, (Map) value);
+      }
+    }
+  }
+
+  public <K, T> T get(AbstractStateDescriptor descriptor, K key) {
+    String desc = descriptor.getIdentify();
+    if (descriptor.getDescType() == DescType.value) {
+      if (this.valueStateMngMap.containsKey(desc)) {
+        return (T) valueStateMngMap.get(desc).get((String) key);
+      }
+    } else if (descriptor.getDescType() == DescType.list) {
+      if (this.listStateMngMap.containsKey(desc)) {
+        return (T) listStateMngMap.get(desc).get((String) key);
+      }
+    } else if (descriptor.getDescType() == DescType.map) {
+      if (this.mapStateMngMap.containsKey(desc)) {
+        return (T) mapStateMngMap.get(desc).get((String) key);
+      }
+    }
+    return null;
+  }
+
+  @Override
+  public void finish(long checkpointId) {
+    for (Entry<String, ValueStateFacade> entry : valueStateMngMap.entrySet()) {
+      entry.getValue().finish(checkpointId);
+    }
+    for (Entry<String, ListStateFacade> entry : listStateMngMap.entrySet()) {
+      entry.getValue().finish(checkpointId);
+    }
+    for (Entry<String, MapStateFacade> entry : mapStateMngMap.entrySet()) {
+      entry.getValue().finish(checkpointId);
+    }
+  }
+
+  @Override
+  public Object commit(long checkpointId, Object state) {
+
+    for (Entry<String, ValueStateFacade> entry : valueStateMngMap.entrySet()) {
+      entry.getValue().commit(checkpointId, entry.getValue().getValueState());
+    }
+    for (Entry<String, ListStateFacade> entry : listStateMngMap.entrySet()) {
+      entry.getValue().commit(checkpointId, entry.getValue().getListState());
+    }
+    for (Entry<String, MapStateFacade> entry : mapStateMngMap.entrySet()) {
+      entry.getValue().commit(checkpointId, entry.getValue().getMapState());
+    }
+    return null;
```

Comment:
Why we return `null` here? If we don't want a return value, use `void`, otherwise `Optional` for nullable case.

Reply:
I'll fix this using void